tests/test_server_proto.py

```diff
@@ -2178,6 +2178,40 @@ async def test_server_proto_ddlprop_01(self):
                             DROP ROLE ddlprop01;
                         ''')
 
+    @unittest.skipUnless(devmode.is_in_dev_mode(),
+                         'the test requires devmode')
+    async def test_server_adjacent_database_propagation(self):
+        if not self.has_create_database:
+            self.skipTest('create database is not supported by the backend')
+
+        conargs = self.get_connect_args()
+
+        server_args = {}
+        if self.backend_dsn:
+            server_args['backend_dsn'] = self.backend_dsn
+        else:
+            server_args['adjacent_to'] = self.con
+        async with tb.start_edgedb_server(**server_args) as sd:
+
+            await self.con.execute('''
+                CREATE DATABASE test_db_prop;
+            ''')
```

Comment:
Can you also add a test for the "a database changes" case?

Reply:
Not sure what you mean by "a database changes" here

Reply:
From code:

```
        # Triggered by a postgres notification event 'database-changes'
        # on the __edgedb_sysevent__ channel
```

Reply:
Well, that's what this test does

Reply:
OK, I got confused with the naming and by the presence of `_on_remote_database_deletion` -- what's it for?

Reply:
Dead code.